### PR TITLE
Settings manager

### LINKED ISSUE #84: Add settings manager
Wee need a way to change several settings values without compiling and deploying.
An example of this is the setup info for PostgreSQL database.

### PR DESCRIPTION
**Pull request by otratsiuk**
closes #84 
Settings class provides a way to read and write persistent settings.
Currently this class is not used but has been tested on TextEdit:

image
image

image
image

### CODE REVIEW

**Comment by Svidunovich** [comment]
Plz add documentation about this class

File: Client/Client.Qt/Settings.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include "Settings.hpp"
+
+Settings::Settings()
+{
+    settings = std::make_unique<QSettings>();
+}
+
+std::unique_ptr<Settings> Settings::instance = nullptr;
+
+Settings &Settings::getInstance()
```
Comment (Svidunovich) [inline Client/Client.Qt/Settings.cpp:10]: in this place should be Settings&, not Settings &

File: Client/Client.Qt/Settings.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+public:
+    static Settings &getInstance();
+    ~Settings(){};
+    void writeSettings();
+    void setFontSize(int size);
+    int getFontSize();
+
+private:
+    int fontSize;
```
Comment (Svidunovich) [inline Client/Client.Qt/Settings.hpp:24]: fix to _fontSize
(if my memory serves me, it should be)

**Review by Svidunovich — CHANGES_REQUESTED** [review]

File: Client/Client.Qt/Settings.cpp (hunk 1 above)
Comment (Orange-hanter) [reply Client/Client.Qt/Settings.cpp]: @Svidunovich сlarify why such rules

Comment (otratsiuk) [reply Client/Client.Qt/Settings.cpp]: Thanks. Found it on wiki, point 25
https://github.com/f1nal3/Juniorgram/wiki/%D0%9A%D0%B0%D0%BA-%D0%BF%D0%B8%D1%81%D0%B0%D1%82%D1%8C-%D0%BA%D0%BE%D0%B4-%D0%BD%D0%B0-CXX

Comment (Svidunovich) [reply Client/Client.Qt/Settings.cpp]: image

**Review by Svidunovich — APPROVED** [review]

**Review by GuskoMaksim — APPROVED** [review]

**Pull request merged**